compiler/optimizations/optimizeOnClauses.cpp

```diff
@@ -541,8 +542,11 @@ static void addRunningTaskModifiers(void) {
       if (fn->hasFlag(FLAG_FAST_ON) == false) {
         CallExpr* call = findRealOnCall(fn);
         SET_LINENO(call);
-        call->insertBefore(new CallExpr(gChplIncRunningTask));
-        call->insertAfter(new CallExpr(gChplDecRunningTask));
+        if (visited.count(call) == 0) {
```

Comment:
Can we change the call variable above so it has a different name from the loop index variable? (It's unnecessarily confusing, as is, IMO).

Also, where are we doing the increment for non-blocking on statements?

Reply:
> Can we change the call variable above so it has a different name from the loop index variable? (It's unnecessarily confusing, as is, IMO).
Yeah, updated. 

> Also, where are we doing the increment for non-blocking on statements?

Here, this modifies the running task counter for the outlined function for all non-fast on stmts.

Reply:
Right, thanks. I was reading FLAG_ON_BLOCK and thinking that meant "blocking on" but it just means "on statement { } ".